### PR TITLE
ci: fix docker

### LINKED ISSUE #1433: fix: docker CI
In the docker file install latest version of tailcall globally using curl as mentioned here https://tailcall.run/docs/getting_started/#curl

Currently docker file is starting the server drop that.
 
Fix the CI to have versioning, right now version is set as main
Check the details of published versions here. Latest is still pointing to v0.18.0
https://github.com/tailcallhq/tailcall/pkgs/container/tailcall%2Ftc-server

### PR DESCRIPTION
**Pull request by neo773**
This sets the tags explicitly using [raw values](https://github.com/docker/metadata-action?tab=readme-ov-file#typeraw)

**Issue Reference(s):**  
Fixes #1433

/claim #1433

## Summary by CodeRabbit

- **New Features**
	- Improved CI workflow to automate assigning the application version and enhance Docker image tagging logic.
- **Refactor**
	- Streamlined the build process by transitioning from Rust to Ubuntu, simplifying installation procedures.

### CODE REVIEW

**Review by ezhil56x — COMMENTED** [review]
Change PR title to `ci: fix docker` for the checks to pass

File: Dockerfile [hunk 1]
```diff
@@ -1,36 +1,4 @@
...
-
-# Copy necessary files from the builder stage
-COPY --from=builder /prod/target/release/tailcall /bin
-COPY --from=builder /prod/jsonplaceholder.graphql /jsonplaceholder.graphql
-
-ENV TAILCALL_LOG_LEVEL=error
-CMD ["/bin/tailcall", "start", "jsonplaceholder.graphql"]
+FROM ubuntu:latest
+RUN apt-get update && apt-get install -y curl
+RUN curl -sSL https://raw.githubusercontent.com/tailcallhq/tailcall/master/install.sh | bash -s -- v0.55.0
```
Comment (amitksingh1490) [inline Dockerfile:3]: version shouldn't be hardcoded here.

Comment (neo773) [reply Dockerfile]: @amitksingh1490 Fixed

**Comment by amitksingh1490** [comment]
@neo773 can you send a screen recording where you have tested this

**Comment by tusharmath** [comment]
Could you please provide some evidence that it's working, @neo773?

**Comment by amitksingh1490** [comment]
@neo773 
 I am getting error
  ERROR [3/3] RUN curl -sSL https://raw.githubusercontent.com/tailcallhq/tailcall/master/install.sh | bash -s                                              2.7s
------
 > [3/3] RUN curl -sSL https://raw.githubusercontent.com/tailcallhq/tailcall/master/install.sh | bash -s:
1.366 bash: line 11: jq: command not found
------
Dockerfile:3
--------------------
   1 |     FROM ubuntu:latest
   2 |     RUN apt-get update && apt-get install -y curl
   3 | >>> RUN curl -sSL https://raw.githubusercontent.com/tailcallhq/tailcall/master/install.sh | bash -s
   4 |     ENV PATH="${PATH}:~/.tailcall/bin"
   5 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c curl -sSL https://raw.githubusercontent.com/tailcallhq/tailcall/master/install.sh | bash -s" did not complete successfully: exit code: 127

**Review by amitksingh1490 — APPROVED** [review]

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**